mono/metadata/w32process.c

```diff
@@ -561,9 +595,9 @@ ves_icall_System_Diagnostics_Process_GetModules_internal (MonoObjectHandle this_
 	return_val_if_nok (error, NULL_HANDLE_ARRAY);
 
 	for (i = 0; i < module_count; i++) {
-		if (mono_w32process_module_get_name (process, mods [i], modname, MAX_PATH)
-			 && mono_w32process_module_get_filename (process, mods [i], filename, MAX_PATH))
-		{
+		modname = mono_w32process_module_get_name (process, mods [i], &modname_len);
+		filename = mono_w32process_module_get_filename (process, mods [i], &filename_len);
+		if (modname && filename) {
```

Comment:
If modname is null, should you call get_filename?

Reply:
I don't think it makes much of a difference, given that's not a common scenario.